### DISCUSSION TITLE
What is a collection

### DISCUSSION THREAD

**Discussion by eeriemyxi** [lexbor/lexbor#200] [Q&A] [answered]
I am new to C/C++ and to learn them I am trying to parse [Wiktionary](https://en.wiktionary.org/).

I am have got some things working [here](https://github.com/eeriemyxi/wiktionary-cli/blob/main/src/impl/parse_word_page.cpp). Mainly I managed to use the CSS selector to target an element with the ID "#English".

Now I am trying to traverse the tree from the grandparent of that element. I found `lxb_dom_node_simple_walk` and it is close to what I am looking for, but it is recursive. My goal is to travel on a constant level of the tree. I then tried something else, namely `lxb_dom_node_by_tag_name` as I am trying to target the H3 elements. But it seems to accept a collection and I am not sure how to deal with that. I tried making a collection but I didn't manage to make one that is empty. I tried `lxb_dom_collection_make(NULL, 128)` but I got segmentation fault when I tried to get its length after making a call to `lxb_dom_node_by_tag_name`. Could somebody explain to me a collection's use case?

**Comment by lexborisov** [comment, marked as answer]
Hi @eeriemyxi 

Function `lxb_dom_node_simple_walk()` does not use recursion to walk the tree.
Check out [the example](https://github.com/lexbor/lexbor/blob/master/examples/lexbor/html/elements_by_tag_name.c) and it will become clear to you how to use this.

**Reply by eeriemyxi** [reply]
Hi. After some trial I managed to get this to work:
```cpp
ParseWordPage::english_main_header_collection = lxb_dom_collection_make(&this->document->dom_document, 128);
lxb_dom_node_by_tag_name(initial_node->parent, this->english_main_header_collection, (const lxb_char_t *) "h3", 2);
```
Thanks!
